api/capi/src/tensor_filter_single.c

```diff
@@ -299,25 +303,49 @@ g_tensor_filter_single_invoke (GTensorFilterSingle * self,
     }
   }
 
-  if (self->allocate_in_invoke == FALSE) {
+  /* set output tensors for given params */
+  _out = output;
+
+  if (self->allocate_in_invoke) {
+    if (!allocate) {
+      /**
+       * @todo how can we remove memcpy if output data is already allocated
+       * single-shot should fill the output data, but sub-plugin allocates new memory.
```

Comment:
If the filter allocates in invoke unconditionally, there is no way to skip memcpy if allocate==TRUE :)